src/app/loadout-builder/process-worker/process-utils.ts

```diff
@@ -600,74 +443,89 @@ function exploreAutoModsSearchTree(
     totalModEnergyCapacity,
   );
+
   if (!picks) {
+    // If we can't hit the target stats with the current exploration stats, we
+    // can't do anything.
     return undefined;
   }
 
-  let bestResult: SearchResult = {
-    depth,
-    picks,
-  };
-
-  // The cost to get to the next tier has one dimension:
-  // * the cost of individual mods (via cheaperStatRelations)
-  const previousStatHashes: number[] = [];
+  // TODO: We need to hit minimums before we hit maximums, so we should
 
-  for (; statIndex < setStats.length; statIndex++) {
-    if (explorationStats[statIndex] >= maxAddedStats[statIndex]) {
-      continue;
+  for (let i = 0; i < explorationStats.length; i++) {
+    if (maxAddedStats[i] <= 0) {
+      continue; // No need to boost this stat
     }
 
-    // Dominance check: If an earlier-explored (=higher-priority) branch needs fewer stat points
-    // to the next tier AND doesn't have more expensive mods than this current one, we don't even need to
-    // look at this branch.
-    if (
-      previousStatHashes.some((previousSubtree) =>
-        info.autoModOptions.cheaperStatRelations[statIndex].includes(previousSubtree),
-      )
-    ) {
-      continue;
+    let candidatePick: ModsPick[] | undefined;
+    const originalExplorationStat = explorationStats[i];
+    if (i === 0) {
+      // Hail mary, try for the best stat boost we can get for this stat.
+      explorationStats[i] = maxAddedStats[i];
+      candidatePick = chooseAutoMods(
+        info,
+        explorationStats,
+        numArtificeMods,
+        remainingEnergyCapacities,
+        totalModEnergyCapacity,
+      );
+
+      if (candidatePick) {
+        // We can hit this stat with the current exploration stats, so try to
+        // increase it.
+        picks = candidatePick;
+        continue; // No need to search for this stat any further
+      }
     }
 
-    const subTreeStats = explorationStats.slice();
-    // Now that tiers no longer matter (since Edge of Fate), we consider any
-    // stat point increase a "tier". This should be a short-term change -
-    // ideally we'd reconsider all these algorithms to see if they could be
-    // simplified now that the tier concept is gone.
-    subTreeStats[statIndex] += 1;
-
-    const explorationResult = exploreAutoModsSearchTree(
-      info,
-      items,
-      setStats,
-      subTreeStats,
-      maxAddedStats,
-      numArtificeMods,
-      remainingEnergyCapacities,
-      totalModEnergyCapacity,
-      statIndex,
-      depth + 1,
-    );
-    // Is this a better solution than what we already have?
-    if (explorationResult && bestResult.depth < explorationResult.depth) {
-      bestResult = explorationResult;
+    // Binary search for the best stat boost we can get for this stat.
+    let lastGoodCandidatePick: ModsPick[] | undefined = undefined;
+    let lastGoodCandidatePickExplorationStat = 0;
+    let minBoost = artificeStatBoost;
+    let maxBoost = maxAddedStats[i] - 1;
```

Comment:
I suspect there's a lot we can do to optimize this - choosing tighter bounds to begin with, or picking only boost amounts that make sense given the possible combination of remaining mods.